pkg/apis/core/validation/validation_test.go

```diff
@@ -5865,6 +5865,106 @@ func TestValidatePodDNSConfig(t *testing.T) {
 	}
 }
 
+func TestValidatePodReadinessGates(t *testing.T) {
+	podReadinessGatesEnabled := utilfeature.DefaultFeatureGate.Enabled("PodReadinessGates")
+	defer func() {
+		// Restoring the old value.
+		if err := utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("PodReadinessGates=%v", podReadinessGatesEnabled)); err != nil {
+			t.Errorf("Failed to restore PodReadinessGates feature gate: %v", err)
+		}
+	}()
+	if err := utilfeature.DefaultFeatureGate.Set("PodReadinessGates=true"); err != nil {
+		t.Errorf("Failed to enable PodReadinessGates feature gate: %v", err)
+	}
+
+	successCases := [][]core.ReadinessGate{
+		// no gate
+		{},
+		// one readiness gate
+		{
+			{
+				ConditionType: core.PodConditionType("example.com/condition"),
+			},
+		},
+		// two readiness gates
+		{
+			{
+				ConditionType: core.PodConditionType("example.com/condition1"),
+			},
+			{
+				ConditionType: core.PodConditionType("example.com/condition2"),
+			},
+		},
+	}
+	for _, tc := range successCases {
+		if errs := validateReadinessGates(tc, field.NewPath("field")); len(errs) != 0 {
+			t.Errorf("expected success: %v", errs)
+		}
+	}
+
+	errorCases := [][]core.ReadinessGate{
+		{
+			{
+				ConditionType: core.PodConditionType("invalid/condition/type"),
+			},
+		},
+	}
+	for _, tc := range errorCases {
+		if errs := validateReadinessGates(tc, field.NewPath("field")); len(errs) == 0 {
+			t.Errorf("expected failure for %v", tc)
```

Comment:
nit: might be more clear to have tc.desc and log it with error

Reply:
or at least `[%d]` with the testcase index